pkg/retry/retry.go

```diff
@@ -49,6 +53,8 @@ func IfNecessary(ctx context.Context, operation func() error, options *Options)
 	return err
 }
 
```

Comment:
```suggestion

// IsErrorRetryable makes a HEURISTIC determination whether it is worth retrying upon encountering an error.
// That heuristic is NOT STABLE and it CAN CHANGE AT ANY TIME.
// Callers that have a hard requirement for specific treatment of a class of errors should make their own check
// instead of relying on this function maintaining its past behavior.
func IsErrorRetryable…
```